### PR TITLE
SNOW-1486524: Upgrade Modin version to 0.28.3 (pins numpy<2)

### PR DESCRIPTION
**Pull request by sfc-gh-dpetersohn**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1486524

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

Upgrade Modin version to 0.28.3 (pins numpy<2)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 MODIN_REQUIREMENTS = [
     *PANDAS_REQUIREMENTS,
     f"modin{MODIN_DEPENDENCY_VERSION}",
+    "pandas==2.2.1",
```
Comment (sfc-gh-nkrishna) [inline setup.py:38]: If Modin installs pandas as well, I don't believe we need the 2.2.1 pin here?

File: CHANGELOG.md [hunk 2]
```diff
@@ -49,6 +49,7 @@
 - Added support for `DataFrame.expanding` and `Series.expanding` for aggregations `count`, `sum`, `min`, `max`, `mean`, `std`, and `var` with `axis=0`.
 - Added support for `DataFrame.rolling` and `Series.rolling` for aggregation `count` with `axis=0`.
 - Added support for `Series.str.match`.
+- Upgrade supported Modin version to 0.28.3
```
Comment (sfc-gh-nkrishna) [inline CHANGELOG.md:52]: I would say this doesn't need a CHANGELOG

**Review by sfc-gh-nkrishna — COMMENTED** [review]
Are there any docs that need updating?

File: setup.py (hunk 1 above)
Comment (sfc-gh-joshi) [reply setup.py:38]: Modin 0.28.1 was a release we specially requested, which pins the patch version 2.2.1; modin 0.28.3 only pins minor version 2.2.x. I don't think 2.2.3 (the latest available version) has any breaking changes, but it's not available in snowflake anaconda and might cause problems if we don't pin here.

**Pull request closed without merging**